CMake/resolve_dependency_modules/icu.cmake

```diff
@@ -75,7 +75,7 @@ foreach(component ${icu_components})
   set_target_properties(
     ICU::${component}
     PROPERTIES IMPORTED_LOCATION ${ICU_${component}_LIBRARY}
-               INTERFACE_INCLUDE_DIRECTORIES ${ICU_INCLUDE_DIRS})
+               INTERFACE_SYSTEM_INCLUDE_DIRECTORIES ${ICU_INCLUDE_DIRS})
```

Comment:
@assignUser Do you remember the reason of this change?

This makes `ICU::uc` not propagate the headers to dependent targets like `re2`.

Reply:
> Projects may also get and set the property directly, but must be aware that adding directories to this property does not make those directories used during compilation.

[docs](https://cmake.org/cmake/help/latest/prop_tgt/INTERFACE_SYSTEM_INCLUDE_DIRECTORIES.html)

Hm seems we have to set both properties? This has been in there a long time, where are you seeing the error?

Reply:
the reason to use it is to avoid warnings in icu code tripping -Werror and blowing up a build